client/components/DataVis.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import {
+  VictoryLine,
+  VictoryChart,
+  VictoryAxis,
+  VictoryTheme,
+  VictoryStack,
+} from "victory";
+import { getUserData, getAllData } from "../store/surveyDataSet";
+import moment from "moment";
+import PainAreaChart from "./PainAreaChart";
+
+export default (props) => {
+  const dispatch = useDispatch();
+  const [surveyData, setSurveyData] = useState([]);
+  const userId = useSelector((state) => state.auth.id);
+  const dataSet = useSelector((state) => state.dataSet);
+
+  const sortedSet = dataSet.sort((a, b) => {
+    return a.id - b.id;
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (userId) {
+        console.log("USERID", userId);
+        await dispatch(getUserData(userId));
```

Comment:
This doesn't need to be inside an async function. The dispatch isn't something that gets awaited, it just tells redux to run your thunk